check_changed.sh

```diff
@@ -16,9 +16,39 @@ then
     exit 0
 fi
 
-cd "$IMAGE_NAME-docker"
 
-CHANGED_FILES=$(git diff --name-status HEAD~1...HEAD .)
+CHANGED_FILES=$(git diff --name-status HEAD~1...HEAD "$IMAGE_NAME-docker")
+
+
+if [ "$IMAGE_NAME" == "theia" ]
+then
+    if [ -z "$CHANGED_FILES" ]
+    then
+        # there were no changes in theia-docker
+        
+        # we want to build the theia-docker image also
+        # in the case that there were changes in other
+        # non theia*-docker folders
+        # eg. build scripts
+    
+        CHANGED_FILES_NON_DOCKER=$(git diff --name-status HEAD~1...HEAD .)
+        while read -r line; do
+            # the output of the git diff is of the form "M    theia-somename-docker/xyz"
+            if [[ $line =~ ^[[:space:]]*.[[:space:]]+theia(.)*-docker(.)+ ]]; then
+                # there was a change to an theia*-docker folder (not theia-docker)
+                # this will be picked up by another build
+                :
+            else
+                # there was a change in a non theia*-docker folder
+                # build theia-full-docker to be sure no utility scripts are broken
```

Comment:
The comment should be updated as well correct?
`# build theia-docker ...`

The same is true for the PR title.